### ISSUE TITLE
experiment: progress-shaped-objective control arm for residual-BC (designed ablation, unpark condition of #1475)

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#6951]
## Current issue-planning review — 2026-09-15

**Audit disposition:** `blocked`. The current audit classifies the issue as blocked; its prerequisite, evidence, or triage gate is incomplete.

**First action:** experiment: progress-shaped-objective control arm for residual-BC (designed ablation, unpark condition of #1475)

### Current bounded plan

1. Inspect the named owners and dependencies before editing; keep the live issue scope and fail-closed evidence boundary intact.
2. Run the named focused validation, record exact output and remaining blockers, and keep fallback, degraded, unavailable, and inconclusive outcomes explicit.

### Files and owners to inspect

- No repository path was identified in the live body; establish the canonical owner before editing.

### Acceptance and validation

- Demonstrate the stated outcome from the live issue with a focused, reproducible check.
- Keep any missing prerequisite or inconclusive result explicit rather than treating it as success.

### Current dependencies and issue-text corrections

- No supplied individual draft existed for this issue in the handoff package; this is a conservative plan derived from the refreshed live body, and the historical issue text is preserved below.
- Live issue references to verify: https://github.com/ll7/robot_sf_ll7/issues/1475; https://github.com/ll7/robot_sf_ll7/issues/2445
- Keep current out-of-scope, ownership, provenance, rights, compute, and claim boundaries intact. Do not close the issue merely because this plan was posted.

### Lifecycle and claim boundary

- Current state labels: `state:blocked`, `state:blocked-no-code-slice`, `state:parked`.
- The issue remains open. No closure, compute submission, release/publication, evidence deletion, or scientific-claim promotion is authorized by this review.
- This review does not request a lifecycle transition; existing assignees, ownership, valid priority labels, and unrelated labels are preserved.
- Proposed non-lifecycle label additions applied with this review: `none`.

**Evidence note:** Full live issue body and complete available comment thread were reviewed during the 2026-09-15 refresh. This is planning metadata, not implementation or benchmark evidence; historical measurements remain historical.

## Background

The residual-BC lane (#1475, parked 2026-07-16) banked a replicated negative result: pure behavior-cloning of a bounded ORCA-residual policy fails to produce a forward-progressing policy — 5 independent smoke attempts (jobs 12670/12672/12749/12913/13034) all ended `success_rate=0.0 / timeout_low_progress` with every infra layer verified working. The failure is objective-level (no progress shaping), consistent with the `revise_residual_objective` ruling (#2445). Five null smokes are anecdotal grade; #1475's stated unpark condition is a designed experiment.

## Designed experiment (this issue)

A two-arm ablation that isolates the objective:

- **Arm A (control):** the original residual-BC objective, unchanged.
- **Arm B:** identical setup plus an explicit progress-shaped objective term (per the #2445 direction).
- Multi-seed (standard evaluation seed policy, not single-seed smokes), same scenario set, same infra, full provenance manifests.

**Promotion criteria:** Arm B produces a forward-progressing policy where Arm A replicates the null result → the objective-level attribution is established at designed-ablation grade. Any other outcome is also informative (records that the objective was not the binding factor).

## Status

Maintainer-requested follow-up (2026-08-11). **Queued, not authorized to launch:** scheduling stays behind the current analysis/admission backlog unless the maintainer elevates it explicitly.

Refs: #1475 (parked lane + unpark condition), #2445 (objective ruling).

```yaml
schema: goal_autopilot_preparation.v1
repository: ll7/robot_sf_ll7
issue: 6951
source_body_sha256: 1d423fcdec531ec404772a08264b43f0c9817306e6c2d0601a7d7dbad291fff1
source_comments_sha256: 
audit_schema: open_issue_contract_audit.v1
audit_digest: 1c191411793929640684e0827d763f3a2009a8682ab98dcefa8c2ded3e54f4ea
audit_classification: state_conflict
next_action: reconcile_state_labels
authority: lifecycle_owner
execution_mode: lifecycle
preferred_worker: MaxRunner
expected_pr_runner_label: runner:max
implementation_admitted: False
state_ready_change_proposed: False
mutation_batch: open-issues-20260830-03
```

This packet is preparation evidence only. It never overrides live labels, exact claim state, branch state, typed dependencies, domain gates, compute authority, release authority, or scientific evidence rules.

**Comment by ll7** [comment]
factory-v2 decision brief: {"category": "scientific_direction", "question": "Issue #6951 \u2014 the designed two-arm ablation (Arm A: unchanged residual-BC objective; Arm B: plus an explicit progress-shaped objective term) that is the maintainer-stated unpark condition of parked #1475 \u2014 is explicitly held: the body records 'Queued, not authorized to launch: scheduling stays behind the current analysis/admission backlog unless the maintainer elevates it explicitly' (2026-08-11); its only label is routing:needs-compute ('excluded from implementation dispatch'); it carries no state:ready, which CONTRIBUTING.md makes the sole dispatch signal. Its deliverable is a multi-seed SLURM campaign of both arms plus promotion-criteria classification, and Arm B's concrete objective term is not named \u2014 the #2445 decision record requires a named objective redesign for a valid reopen, and the existing progress-named runtime candidate orca_residual_guarded_ppo_progress_v1 already produced null smoke job 12913. Owner decision needed: (a) elevate and authorize launch explicitly, and name (or delegate naming of) the Arm B progress-shaping term and the quantitative promotion thresholds; (b) authorize only the local no-compute implementation slice first; or (c) keep the issue queued behind the backlog? Recommended engineering action for (b): one bounded PR adding a progress-weighted BC objective variant (per-transition expert-action NLL weighted by normalized per-step route progress, with Arm A as the identical pipeline under uniform weights) in scripts/training/pretrain_from_expert.py and robot_sf/training/, two-arm configs under configs/training/orca_residual/ pinned to eval seeds 111/112/113 (configs/benchmarks/seed_sets_v1.yaml) and scenario planner_sanity_simple, an experiment record under experiments/, and tests under tests/training/ \u2014 with no SLURM submission, no campaign execution, and no writes under docs/context/evidence/ or any benchmark path (that slice triggers no Domain-Aware Approval gate)."}

**Comment by ll7** [comment]
ll7/robot_sf_ll7#6951: b

Decision resolved: authorize the bounded local, no-compute implementation slice first. No SLURM job, training campaign, or evidence claim is authorized by this decision.

Five independent `success_rate=0.0` / `timeout_low_progress` smokes make another unchanged run uninformative. Also, the existing `orca_residual_guarded_ppo_progress_v1` revision changed the residual envelope and runtime progress-comparison margin; it did not add a progress-shaped behavioural-cloning loss. A local implementation is therefore the right next step: it makes the new mechanism reviewable and prevents objective details from being chosen after results are seen.

Use this implementation contract:

- add one named, configurable progress-weighted expert-action negative-log-likelihood objective;
- Arm A must use the same dataset, batching, initialization, epochs, and code path with uniform transition weights;
- Arm B should derive a signed per-step signal from reduction in remaining route length, normalize it by a declared scale, and apply a bounded positive weight such as `clip(1 + lambda * progress, min_weight, max_weight)` so stalled/regressing samples are not silently discarded;
- fail closed when the dataset lacks aligned route-progress provenance; do not substitute episode success, reward, future outcome labels, or world-coordinate displacement as hidden proxies;
- expose the objective name, `lambda`, normalization, bounds, dataset digest, and random seeds in config and the run manifest;
- add focused unit tests for Arm-A equivalence, weighting direction/bounds, malformed or missing progress data, and deterministic config/manifest serialization;
- stage matched Arm A/B configs under `configs/training/orca_residual/` and a no-launch experiment record. Seeds 111/112/113 and `planner_sanity_simple` may be preregistered, but must not be executed yet.

After the local PR merges, return for a separate compute decision that freezes the campaign values and paired success/progress thresholds. Keep #1475 parked until then; this decision satisfies only the concrete-proposal part of its unpark condition, not the compute-lane or smoke-to-nominal gates.

Confidence: 94%. This is the cheapest step that tests a genuinely new objective while preserving a clean designed-ablation boundary.

**Comment by ll7** [comment]
Reopened after PR #6997 merged the authorized local, no-compute implementation slice.

The issue remains the owner of the separate compute decision: dataset provenance, campaign authorization, paired seeds, and success/progress thresholds are still not frozen or launched. The merge did not produce empirical evidence, benchmark results, or a paper-facing claim.

The implementation PR explicitly stated that it does not close this issue; the issue is retained for the follow-up decision and any authorized paired experiment.

**Comment by ll7** [comment]
Added the `routing:needs-compute` label so autonomous issue selection does not treat the merged local implementation slice as permission to launch the paired experiment.

The remaining work is still the explicitly separate compute decision described in the issue: freeze dataset provenance, seeds, and success/progress thresholds before any campaign is authorized.

**Comment by ll7** [comment]
Authorized bounded local-CPU diagnostic completed for the #6951 Arm A/B ablation.

- Protocol: three expert episodes; dataset SHA-256 `aa70b611875be2b6b67001e063bcf558ff00eff4ef398a4da498f33df8a07a09`; training seeds 111/112/113; deterministic evaluation on `planner_sanity_simple` for the same three seeds.
- Arm A (uniform): 0/9 successes, 9/9 collisions, mean route-length reduction 3.254 m.
- Arm B (progress-weighted): 0/9 successes, 9/9 collisions, mean route-length reduction 3.489 m.
- Paired Arm B minus Arm A route-reduction difference: +0.235 m; cell-level bootstrap 95% interval [+0.191, +0.283] m.
- Execution was native CPU with no fallback or degraded rows.

This is diagnostic-only smoke evidence. The promotion condition was not met, so #1475 stays parked and #6951 stays open; no released default or paper-facing claim changes. Raw datasets, checkpoints, and logs remain worktree-local. The tracked synthesis is [`docs/context/evidence/issue_6951_progress_shaped_ablation/README.md`](https://github.com/ll7/robot_sf_ll7/blob/main/docs/context/evidence/issue_6951_progress_shaped_ablation/README.md).

Next step: correct or explicitly freeze the observation-history contract, then run a properly sized paired campaign with durable dataset/checkpoint retrieval metadata and maintainer-frozen success/progress thresholds.

**Comment by ll7** [comment]
### Blocked-queue triage (tracking #7067)



```yaml
blocker_class: maintainer
unblock_condition: >-
  The maintainer elevates this ablation above the current analysis/admission backlog and authorizes a properly sized paired campaign, after the observation-history contract is corrected or explicitly frozen and success/progress thresholds are maintainer-frozen.
watcher: >-
  `gh issue view 6951 --repo ll7/robot_sf_ll7 --json labels --jq '[.labels[].name]'` no longer contains `state:blocked` / `routing:needs-compute`; maintainer response on this thread
next_check_at: >-
  next maintainer triage pass
last_meaningful_progress_at: "2026-08-13"
```

The 2026-08-13 authorized bounded local-CPU diagnostic ran both arms (Arm A 0/9, Arm B 0/9 successes) and did not meet the promotion condition, so the issue stays open. The remaining blocker is a maintainer decision, not a missing artifact: the body records the experiment as *"queued, not authorized to launch"* pending explicit elevation, and the diagnostic's own next step needs frozen thresholds before a sized campaign can run.

Bookkeeping only — no scope, claim, label, or evidence change is made by this comment, and the issue stays open.

**Comment by ll7** [comment]
## Unblock route — 2026-08-16

**Use the existing result first:** Preserve the completed bounded comparison and its endpoint failure as a negative mechanism result. Do not repeat the same weighted-versus-unweighted run.

**Permitted local work:** Convert the result into a #7029 packet and #7032 mechanism-boundary card, with exact support, collision endpoint, route-progress change, and unresolved observation-history/data-coverage hypotheses.

**Further training remains parked:** Unpark only after a maintainer selects one discriminating hypothesis, freezes a new design, and records why the new run can change the decision.

**Comment by ll7** [comment]
ll7/robot_sf_ll7#6951: accept-negative-diagnostic-and-park

## Author ruling — 2026-08-17

Preserve the completed bounded comparison as a negative diagnostic and park further residual-behavior-cloning training.

Both arms produced 0/9 successes and 9/9 collisions. The progress-weighted arm's approximately +0.235 m paired route-reduction difference is a mechanism signal, but it does not satisfy the success/collision promotion condition and must not be presented as a policy improvement.

Automatic consequences:

- permit only conversion of the existing result into the #7029 interpretation packet and #7032 mechanism-boundary card;
- preserve exact support, collision endpoints, route-progress change, diagnostic evidence tier, and unresolved observation-history/data-coverage hypotheses;
- remove `decision-required`, add `ruled`, and keep the issue parked;
- authorize no repeat run, larger campaign, policy promotion, released-default change, benchmark claim, or paper-facing claim.

Reopen only after the observation-history contract is corrected or explicitly frozen and one new discriminating hypothesis has a preregistered paired design, durable dataset/checkpoint route, fixed success/progress thresholds, compute cap, and stop rule.

**Comment by ll7** [comment]
## Revival contract — 2026-08-19

This designed ablation is parked and not authorized to launch. `state:running` was stale.

Revive first as a preparation-only contract that freezes: the exact Arm A and Arm B objective definitions; source dataset and split aliases; model/checkpoint lineage; scenario and seed roster; progress, timeout, collision, fallback, and degradation predicates; paired analysis unit; minimally important effect or diagnostic-only disposition; artifact plan; resource estimate; and negative/inconclusive outcomes. The preparation slice may add no training result and may submit no compute.

A later execution child requires the completed design packet, one immutable differential fingerprint, concrete durable aliases, domain approval, and fresh compute authorization. An identical retry of the historical residual-BC objective remains prohibited.
